ctc_decoder/beam_search.py

```diff
@@ -89,23 +95,29 @@ def beam_search(mat: np.ndarray, labels: str, beam_width: int = 25, lm: Optional
         for labeling in best_labelings:
 
             # probability of paths ending with a non-blank
-            pr_non_blank = 0
+            pr_non_blank = LOG_ZERO
             # in case of non-empty beam
             if labeling:
                 # probability of paths with repeated last char at the end
-                pr_non_blank = last.entries[labeling].pr_non_blank * mat[t, labeling[-1]]
```

Comment:
to keep the code as uncluttered as possible, I suggest moving the logic which takes care of adding two values together into a separate function, as you need it multiple times and we can get rid of the if / else in the code.
Something like:
```

def log_add(x, y):
    if x == log(0):
        return y
    if y == log(0):
        return x
    return x + y
```